### PR TITLE
docs: add status badge for the periodic cronjob

### PR DESCRIPTION
**Pull request by rexagod**
Additionally, rename it to a more fitting name.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Overview
 
 [![Build Status](https://github.com/kubernetes/kube-state-metrics/workflows/continuous-integration/badge.svg)](https://github.com/kubernetes/kube-state-metrics/actions)
-[![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/kube-state-metrics)](https://goreportcard.com/report/github.com/kubernetes/kube-state-metrics) [![GoDoc](https://godoc.org/github.com/kubernetes/kube-state-metrics?status.svg)](https://godoc.org/github.com/kubernetes/kube-state-metrics)
+[![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/kube-state-metrics)](https://goreportcard.com/report/github.com/kubernetes/kube-state-metrics)
+[![GoDoc](https://godoc.org/github.com/kubernetes/kube-state-metrics?status.svg)](https://godoc.org/github.com/kubernetes/kube-state-metrics)
```
Comment (olivierlemasle) [inline README.md:5]:
```suggestion
[![Go Reference](https://pkg.go.dev/badge/github.com/kubernetes/kube-state-metrics.svg)](https://pkg.go.dev/github.com/kubernetes/kube-state-metrics)
```

**Review by olivierlemasle — COMMENTED** [review]
LGTM, but you might take advantage of this change to update the GoDoc reference.

File: .github/workflows/govulncheck.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: cron
+name: govulncheck
```
Comment (CatherineF-dev) [inline .github/workflows/govulncheck.yml:1]: nit: go-vuln-check

Comment (dgrisonnet) [reply .github/workflows/govulncheck.yml:1]: `govulncheck`  is the name of the binary being used so I am fine with keeping that name

Comment (CatherineF-dev) [reply .github/workflows/govulncheck.yml:1]: Okay!

**Pull request merged**